contracts/BonsaiStarter.sol

```diff
@@ -14,56 +14,36 @@
 
-import {IBonsaiRelay} from "bonsai/IBonsaiRelay.sol";
-import {BonsaiCallbackReceiver} from "bonsai/BonsaiCallbackReceiver.sol";
+import {IRiscZeroVerifier} from "bonsai/IRiscZeroVerifier.sol";
 
-/// @title A starter application using Bonsai through the on-chain relay.
+/// @title A starter application using Bonsai.
 /// @dev This contract demonstrates one pattern for offloading the computation of an expensive
 //       or difficult to implement function to a RISC Zero guest running on Bonsai.
-contract BonsaiStarter is BonsaiCallbackReceiver {
-    /// @notice Cache of the results calculated by our guest program in Bonsai.
-    /// @dev Using a cache is one way to handle the callback from Bonsai. Upon callback, the
-    ///      information from the journal is stored in the cache for later use by the contract.
-    mapping(uint256 => uint256) public fibonacciCache;
-
-    /// @notice Image ID of the only zkVM binary to accept callbacks from.
-    bytes32 public immutable fibImageId;
-
-    /// @notice Gas limit set on the callback from Bonsai.
-    /// @dev Should be set to the maximum amount of gas your callback might reasonably consume.
-    uint64 private constant BONSAI_CALLBACK_GAS_LIMIT = 100000;
-
-    /// @notice Initialize the contract, binding it to a specified Bonsai relay and RISC Zero guest image.
-    constructor(IBonsaiRelay bonsaiRelay, bytes32 _fibImageId) BonsaiCallbackReceiver(bonsaiRelay) {
-        fibImageId = _fibImageId;
-    }
-
-    event CalculateFibonacciCallback(uint256 indexed n, uint256 result);
-
-    /// @notice Returns nth number in the Fibonacci sequence.
-    /// @dev The sequence is defined as 1, 1, 2, 3, 5 ... with fibonacci(0) == 1.
-    ///      Only precomputed results can be returned. Call calculate_fibonacci(n) to precompute.
-    function fibonacci(uint256 n) external view returns (uint256) {
-        uint256 result = fibonacciCache[n];
-        require(result != 0, "value not available in cache");
-        return result;
+contract EvenNumber {
+    /// @notice RISC Zero verifier contract address.
+    IRiscZeroVerifier verifier;
+    /// @notice Image ID of the only zkVM binary to accept verification from.
+    bytes32 imageId;
+    /// @notice Value of the last successfully updtated number.
+    uint256 number;
+
+    /// @notice Initialize the contract, binding it to a specified RISC Zero verifier and guest image.
+    constructor(IRiscZeroVerifier _verifier, bytes32 _imageId) {
+        verifier = _verifier;
+        imageId = _imageId;
+        number = 0;
     }
 
-    /// @notice Callback function logic for processing verified journals from Bonsai.
-    function storeResult(uint256 n, uint256 result) external onlyBonsaiCallback(fibImageId) {
-        emit CalculateFibonacciCallback(n, result);
-        fibonacciCache[n] = result;
+    /// @notice Function logic for processing verified journals from Bonsai.
+    function set(uint256 x, bytes32 postStateDigest, bytes calldata seal) public {
```

Comment:
It's a bit unfortunate that `postStateDigest`, which is a fairly hard to explain item, is included in this interface. I'll have to think about how to deal with that, since I think this needs to be addressed in `IRiscZeroVerifier`